aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -507,7 +508,7 @@ class CuFFTParamsLRUCache {
     // first.
     TORCH_CHECK(new_size >= 0,
              "cuFFT plan cache size must be non-negative, but got ", new_size);
-    TORCH_CHECK(new_size <= CUFFT_MAX_PLAN_NUM,
+    TORCH_CHECK(new_size <= static_cast<int64_t>(CUFFT_MAX_PLAN_NUM),
```

Comment:
Should we just make CUFFT_MAX_PLAN_NUM a int64_t from the beginning? There is a static assert that it is always positive already.

Reply:
I'd support that. Signed types seem to be generally easier to work with; however, I suspect we'll still have inappropriate comparisons. Let me know what works best for you.

Reply:
Given that these values can never really overflow a signed int. I would lean towards whatever reduces the number of casts the most :D
But if you think this is cleaner, I'm happy with that as well.